### ISSUE TITLE
mapping

### ISSUE THREAD

**Issue by NewGraphEnvironment** [NewGraphEnvironment/fish_passage_bulkley_2020_reporting#2]
@smnorris - geopackage is populated [here](https://github.com/NewGraphEnvironment/fish_passage_bulkley_2020_reporting/blob/master/data/fishpass_mapping.gpkg)

not sure the best approach for version control of the project maps hosted on the Hillcrest site.  Using the date stamp of the map as the name of the subfolder one way to go but maybe only necessary on the last version?  Until then we could have a subfolder “NewGraph” for these maps if there are multiple versions of maps on the server.  Not sure.

still interested in using q atlas and the watersheds to build tighter maps for each hab con but think they need to have legends and keymaps .

**Comment by NewGraphEnvironment** [comment]
mapping files are now pulled from geopackage then burned to geojson format in [fishpass_mapping](https://github.com/NewGraphEnvironment/fish_passage_bulkley_2020_reporting/tree/master/data/fishpass_mapping) with [b2f3d0d8](https://github.com/NewGraphEnvironment/fish_passage_bulkley_2020_reporting/commit/b2f3d0d8ec243bf1d0dcbe6f574a5580bdb934bb)

**Comment by smnorris** [comment]
If you convert to WGS84 / EPSG:4326 they will preview nicely in GH

**Comment by NewGraphEnvironment** [comment]
:fire:

**Comment by smnorris** [comment]
We now have habitat model data for Bulkley, and potentially for Morice as well.

Do we want to change the mapping to vary the stream symbol width by habitat classification rather than by gradient? 
ie
narrow=non spawning/rearing habitat
medium=rearing
wide=spawning.

Retain the same accessibility colour scheme.

**Comment by smnorris** [comment]
Of course there is also the option to symbolize stream width by the modelled channel width, but modelled habitat seems more relevant.

**Comment by smnorris** [comment]
Updated spawning/rearing classification should be ready in the next day or so https://github.com/smnorris/bcfishpass/issues/46

**Comment by NewGraphEnvironment** [comment]
I have had a look through but can't seem to find it. Sorry.  Where are the gradient thresholds for the accessibility? I am thinking about steelhead vs. salmon vs. WCT ...

**Comment by smnorris** [comment]
Salmon = 15
Steelhead = 20
WCT = 20

~~This is a good point. The accessibility mapping colours for Bulkley are currently based on steelhead (20%). Symbolizing based on salmon habitat will mean that steelhead accessible streams above a 15% gradient barrier won't show any habitat.~~
My mistake, steelhead habitat is modelled.

**Comment by smnorris** [comment]
FYI the access classification logic is buried in here: https://github.com/smnorris/bcfishpass/blob/main/02_model/sql/model_access.sql

**Comment by NewGraphEnvironment** [comment]
> We now have habitat model data for Bulkley, and potentially for Morice as well.
> 
> Do we want to change the mapping to vary the stream symbol width by habitat classification rather than by gradient?
> ie
> narrow=non spawning/rearing habitat
> medium=rearing
> wide=spawning.
> 
> Retain the same accessibility colour scheme.

We would pick a "species" based on `02_model/data/model_spawning_rearing_habitat.csv` to vary the stream symbol width by habitat classification?

**Comment by smnorris** [comment]
Rather than pick just one species I was thinking of lumping all spawning into one class and all rearing into another.

**Comment by NewGraphEnvironment** [comment]
Sound like a good plan. Let's do it.  

Hoping for some background on the width modelling and some references for habitat requirements from CWF soon :no_mouth: .  I shared my thin zotero modelling folder with you and Nick.  We can keep a copy of our references there if that works for you guys.

**Comment by NewGraphEnvironment** [comment]
Simon. Any chance of a refresh on the maps with Jonathan's barriers now input and an update to the gpkg pushed to data/fishpass_mapping? Shooting for the end of the day to get a draft of the report to CWF and FPTWG (no real morice section yet)...  I doubt anyone will look at it right away so don't think it would matter too much if the maps were updated tomorrow...

**Comment by smnorris** [comment]
Can it wait another couple of days? I am working on the spawn/rear connectivity updates, it would be nice to finish that first: https://github.com/smnorris/bcfishpass/issues/46 

~~The in-stream distance thresholds mean I'm fixing this up so it works properly
https://github.com/smnorris/fwapg/blob/main/sql/functions/FWA_LengthInstream.sql~~
edit - found a way around this so that function isn't getting cleaned. It should probably be removed until it is.

**Comment by NewGraphEnvironment** [comment]
Sure. I was just thinking that I should get a bit more of the methods documented and bring some area tables in to the results anyways. Needs to be done for the Morice anywho. 

Maybe it’s done already but shouldn’t be too hard to add areas of streams by spawn/rear per species right? Will make a lovely table and might even read easier as a graph. Lots of info but I think the reader can handle it and we can tuck it into a leaflet link too so it is tight and available.

I’ll start looking at it tomorrow. Going to have another crack at getting the bcfishpass repo to run for me. 🤞

**Comment by smnorris** [comment]
Areas of streams?  channel width * length? Hasn't been any discussion of that.

**Comment by NewGraphEnvironment** [comment]
Makes sense to me.  I reported area for hab cons for the elk and bulk just using length and measured width from my surveys. In general, a 8m wide stream that is 1k to a waterfall will have waaaay more potential for useable habitat than a stream that is 4m wide and 1k to a waterfall so it is useful to visualize and communicate that I think. Cost per m2 has also been quite enlightening for me.

**Comment by smnorris** [comment]
Ok. There was a good discussion today on how to fudge Coho rearing habitat in wetlands and Sockeye rearing habitat in lakes into linear measurements. Stupidly it never occurred to me to suggest we just report on area for everything - that has been the idea for quite some time and we actually have the data to do it now.

**Comment by smnorris** [comment]
Well off topic here tho - something to file in bcfishpass

**Comment by NewGraphEnvironment** [comment]
I'm going to start a list here to store things related to future runs of the maps unless you think there is a better spot:

- [ ] Add modelled crossings to the maps so they can be used as ids in future field surveys and viewed by readers when referenced in reporting.

**Comment by smnorris** [comment]
Thanks - that is a bug, they should be on there!

**Comment by smnorris** [comment]
The filter in the crossings layer was using stream_crossing_id so NULLs for modelled crossing got filtered out too. 
To fix, the filter that removes PSCIS crossings (that get replaced by confirmations not yet in PSCIS) needs to use aggregated_crossings_id

**Comment by smnorris** [comment]
moving issue from bcfishpass to here: 

>>>https://hillcrestgeo.ca/outgoing/fishpassage/projects/bulkley/FishPassage_093M.102.pdf (Porphyry) doesn't currently want to open in my browser. Other maps do though...

**Comment by smnorris** [comment]
It doesn't open on my desktop either, I'll re-export that one, I don't know what might be wrong.

**Comment by smnorris** [comment]
Regenerated and reposted file. But it looks like it was a file transfer issue, a bunch of them are empty. I'll repost.

**Comment by smnorris** [comment]
Ha. Server is out of space, no wonder it is broken. 🤪

**Comment by NewGraphEnvironment** [comment]
aha.

**Comment by smnorris** [comment]
Fixed but we'll want to minimize how many of these monster pdfs we keep in the archive folders for now. I removed 2021-02-15 to fit these.

**Comment by NewGraphEnvironment** [comment]
you could pull the other archives off too if you want.  From my perspective, the archiving is just for a stable location for readers to land when referred from the report. Those old maps do have the FN sites on there though so I think that some people might be keen to have them.... I have a copy if anyone wants them.

**Comment by smnorris** [comment]
ok - I'll keep them there for now. I may be able to shuffle things around, I have lots of unused space, it just isn't accessible by the web server at the moment.

**Comment by NewGraphEnvironment** [comment]
@smnorris.  Quick question... Where are the widths in `bcbarriers/bcfishpass.streams` for the Morice modelling on the maps currently pulled from? Did you use the Bulkley chan width best fit equation?  Is there another equation specific to the MORR somewhere that you know of? Sounds like the numbers where pretty off and it needs to be revisited but maybe not a bad idea to start with what has already been done....

**Comment by smnorris** [comment]
The db has the measured values from `bcfishpass.channel_width_measured` - not sure about the maps, it may just be gradient

**Comment by smnorris** [comment]
No - maps for `MORR` are the same as `BULK` - habitat model is applied... but only where there is a measured channel width. Good catch, that is going to have to be updated.

**Comment by NewGraphEnvironment** [comment]
right. That's why the symbology is so jumpy on the maps.  Ok.  I will line up something to get some modelled estimates. Working through the guts of the scripts. Awesome stuff. Lots to get my head around. Trying to figure out why our dbs have different results for a few things... I'll let you know if i can't figure it out on my own.

**Comment by NewGraphEnvironment** [comment]
I have used the BULK model as a start and switched the BULK mad switch to cw for fun.  Haven't qa'd anything yet. Just poking at stuff.  👆🏼
